### ISSUE TITLE
improvement for "allowDifferentNoteFileName" alert window and option label

### ISSUE THREAD

**Issue by cleantext** [pbek/QOwnNotes#3001]
Hi and thanks for the awesome text editor! I recently installed qOwnNotes and am quite excited about it.

This actually turned from a bug report to a feature request :-)

### Case

When I right-click on a note and select Rename Note, I get an alert window: 

"If you want to rename your note you have to enable the option to allow the note filename to be different from the headline."

And there is a button that opens up Settings.

I find four user experience issues in this case, in no specific order:

1. The Settings window can be overwhelming for a newcomer.
2. The "allowDifferentNoteFileName" setting can be counter-intuitive\*.
3. The alert window's text is vague.
4. The Settings window differs from traditional UI patterns by listing its sections/tabs in a different order. For example, it does not start with General, etc.

\* Counter-intuitive because, in the majority of cases, settings for how a text editor handles files apply to the whole program and not separate folders. Having years of experience in IT and working with various text editors, I find it unlikely that I'm an outlier in that way of reasoning. Although I appreciate the granularity the per-folder setting provides, now that I know where it is!

### Request

Please provide some hand-holding for the user by providing more clarity in the text of the alert window:

Here's a suggestion.

What is now:
"If you want to rename your note you have to enable the option to allow the note filename to be different from the headline."

Can become:
"To rename your note, enable 'Allow note file name to be different from headline' from Settings -> Note Folders."

Or we can expand a bit more:
"To rename your note, enable 'Allow note file name to be different from headline' from Settings -> Note Folders. Please note this setting is per-folder."

To further expand on my suggestion, here is a label for the option that sounds a bit more fluent:

What is now:
"Allow note file name to be different from headline"

Can become:
"Do not sync note filename with its headline"
\* "note filename" instead of "note file name" so that we don't have one adjective serving two nouns.

Or just:
"Disable automatic file naming"

---

With all that said, maybe I am used to having a Settings option start with the General tab/section. Or maybe I am used to expecting file system options to be application-wide, not per-folder.  Either way, I wasn't met with the usual UI patterns my brain is used to and that threw me off. 

I hope you will consider this change and improvement in the alert, so that it's easier for us to navigate the uniqueness of qOwnNotes.

Thanks once again for all the time you're putting into this project.

**Comment by pbek** [comment]
Thank you for your kind words!

I'm not sure yet about your request...

The whole purpose of QOwnNotes (and the feature set of the first release, 1089 releases ago) is that one is able to create a new file very quickly, with a predetermined filename and to be able to write to that file with the text automatically to be stored. The first line of the file determined the filename, to not need to bother with them manually.

So "filename = headline" is the default. Much later I allowed it to differ, and I did that for note folders individually, because I think needs might be different for each note folder in that regard (e.g. if you want to write articles for a static site generator in a note folder).

So much for the history behind that...

I'm not sure if we need a very detailed description in the dialog about where the setting is if the button already brings you exactly to the right spot for the current note folder. People are overwhelmed by lots of text (like this feature request 😉).

And about the checkbox text:

>  "Do not sync note filename with its headline"

Does everyone know what "sync" is? I'm not sure. 🤔

>  "Disable automatic file naming"

Does this really clear up things? And then all references to that setting would need to be renamed too... ️🤷🏻
